explore/src/main/scala/explore/observationtree/AsterismGroupObsListActions.scala

```diff
@@ -229,42 +236,29 @@ object AsterismGroupObsListActions {
     }
   )
 
-  def targetExistence(
-    targetId: Target.Id
-  )(implicit
-    appCtx:   AppContextIO
-  ) = // can switch appCtx to TransactionalClient when logging not required
-    Action[AsterismGroupsWithObs, Option[TargetGroup]](
-      getter = (agwo: AsterismGroupsWithObs) => agwo.targetGroups.get(targetId),
-      setter = (otg: Option[TargetGroup]) =>
-        (agwo: AsterismGroupsWithObs) =>
-          if (agwo.targetGroups.contains(targetId)) {
-            // Delete - don't need to worry about the asterism groups for delete
-            val newGroups = agwo.targetGroups.removed(targetId)
-            agwo.copy(targetGroups = newGroups)
-          } else
-            // Undelete: Should always have a TargetGroup at this point.
-            otg.fold(agwo) { tg =>
-              val newTargetGroups   = agwo.targetGroups.updated(targetId, tg)
-              val obsInTg           = tg.obsIds
-              // update the asterism groups
-              val newAsterismGroups = agwo.asterismGroups
-                .map { case (_, ag) =>
-                  if (ag.obsIds.toSortedSet.intersect(obsInTg).nonEmpty) ag.addTargetId(targetId)
-                  else ag
-                }
-                .toList
-                .toSortedMap(_.obsIds)
-              agwo.copy(targetGroups = newTargetGroups, asterismGroups = newAsterismGroups)
-            }
+  def targetExistence(targetId: Target.Id, setPage: Option[Target.Id] => IO[Unit])(implicit
+    c:                          TransactionalClient[IO, ObservationDB]
+  ): Action[AsterismGroupsWithObs, Option[TargetWithObs]] =
+    // can switch appCtx to TransactionalClient when logging not required
+    Action[AsterismGroupsWithObs, Option[TargetWithObs]](
+      getter = (agwo: AsterismGroupsWithObs) => agwo.targetsWithObs.get(targetId),
+      setter = (otwo: Option[TargetWithObs]) =>
+        otwo.fold(AsterismGroupsWithObs.targetsWithObs.modify(_.removed(targetId))) { tg =>
+          AsterismGroupsWithObs.targetsWithObs.modify(_.updated(targetId, tg)) >>>
+            AsterismGroupsWithObs.asterismGroups.modify(
+              _.map { case (_, ag) =>
+                if (ag.obsIds.toSortedSet.intersect(tg.obsIds).nonEmpty) ag.addTargetId(targetId)
+                else ag
+              }.toList.toSortedMap(_.obsIds)
+            )
+        }
     )(
-      onSet = (_, otg) =>
-        otg.fold(
-          appCtx.logger.error("Creating targets in AsterismGroupObsListActions not yet supported")
-        )(_ => deleteTarget(targetId).void),
+      // DB creation is performed beforehand, in order to get id
+      onSet =
+        (_, otg) => otg.fold(deleteTarget(targetId) >> setPage(none))(_ => setPage(targetId.some)),
       onRestore = (_, otg) =>
-        otg.fold(deleteTarget(targetId).void) { _ =>
-          undeleteTarget(targetId).void
+        otg.fold(deleteTarget(targetId) >> setPage(none)) { _ =>
```

Comment:
Does the target have to focused for us to delete it? If not, should we just leave the page as it is so we don't unfocus something else? If it was focused and is deleted, that should be fixed elsewhere.